include/SFML/Window/WindowBase.inl

```diff
@@ -0,0 +1,75 @@
+////////////////////////////////////////////////////////////
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2007-2024 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/Window/Event.hpp>
+#include <SFML/Window/WindowBase.hpp> // NOLINT(misc-header-include-cycle)
+
+#include <variant>
+
+
+namespace sf
+{
+namespace priv
+{
+template <typename... Ts>
+struct Overloaded : Ts...
+{
+    using Ts::operator()...;
+};
+template <typename... Ts>
+Overloaded(Ts...) -> Overloaded<Ts...>;
+
+template <typename... Ts>
+struct OverloadedWrapper : Overloaded<Ts...>
+{
+    // By providing our own operator() and forwarding based
+    // on invocability of Overloaded<Ts...> on the concrete type
+    // of the value, we save the user from having to provide
+    // their own catch-all overload if they don't want to
+    template <typename T>
+    void operator()(T&& value) // NOLINT(cppcoreguidelines-missing-std-forward)
+    {
+        if constexpr (std::is_invocable_v<Overloaded<Ts...>, T>)
+            Overloaded<Ts...>::operator()(std::forward<T>(value));
+
+        (void)value;
+    }
+};
+template <typename... Ts>
+OverloadedWrapper(Ts...) -> OverloadedWrapper<Ts...>;
+} // namespace priv
+
+
+////////////////////////////////////////////////////////////
+template <typename... Ts>
+void WindowBase::handleEvents(Ts&&... handlers) // NOLINT(cppcoreguidelines-missing-std-forward)
```

Comment:
Why did you need to disable this clang-tidy check in a couple places?

Reply:
clang-tidy on specific platforms would still complain about missing `std::forward` even though it was there.

Or in other words, clang-tidy was spitting out false positives on certain systems (not all).

https://github.com/SFML/SFML/actions/runs/9714239737/job/26812967949